cli/azd/pkg/input/console.go

```diff
@@ -338,3 +338,71 @@ func GetStepResultFormat(result error) SpinnerUxType {
 	}
 	return formatResult
 }
+
+// A console implementation which output goes only to logs
+// This is used to prevent or stop actions using the terminal output, for
+// example, when calling provision during deploying a service.
+type silentConsole struct {
+}
+
+// Sets the underlying writer for output the console or
+// if writer is nil, sets it back to the default writer.
+func (sc *silentConsole) SetWriter(writer io.Writer) {
+	log.Println("tried to set writer for silent console is a no-op action")
+}
+
+func (sc *silentConsole) GetFormatter() output.Formatter {
+	return nil
+}
+
+func (sc *silentConsole) IsUnformatted() bool {
+	return true
+}
+
+// Prints out a message to the underlying console write
+func (sc *silentConsole) Message(ctx context.Context, message string) {
+	log.Println(message)
+}
+
+func (sc *silentConsole) MessageUxItem(ctx context.Context, item ux.UxItem) {
+	sc.Message(ctx, item.ToString(""))
+}
+
+func (sc *silentConsole) ShowSpinner(ctx context.Context, title string, format SpinnerUxType) {
+	log.Printf("request to show spinner on silent console with message: %s", title)
+}
+
+func (sc *silentConsole) StopSpinner(ctx context.Context, lastMessage string, format SpinnerUxType) {
+	log.Printf("request to stop spinner on silent console with message: %s", lastMessage)
+}
+
+// Prompts the user for a single value
+func (sc *silentConsole) Prompt(ctx context.Context, options ConsoleOptions) (string, error) {
+	log.Panic("Can't call Prompt from a silent console.")
+	return "", nil
+}
+
+// Prompts the user to select from a set of values
+func (sc *silentConsole) Select(ctx context.Context, options ConsoleOptions) (int, error) {
+	log.Panic("Can't call Select from a silent console.")
+	return 0, nil
+}
+
+// Prompts the user to confirm an operation
+func (sc *silentConsole) Confirm(ctx context.Context, options ConsoleOptions) (bool, error) {
+	log.Panic("Can't call Confirm from a silent console.")
+	return false, nil
+}
+
+func (sc *silentConsole) GetWriter() io.Writer {
+	return nil
+}
+
+func (sc *silentConsole) Handles() ConsoleHandles {
+	return ConsoleHandles{}
+}
+
+// Creates a silent console
+func NewSilentConsole() Console {
```

Comment:
I wonder if it is possible for us to factor things such that we don't have to export this type?  It feels like is very tied to a specific problem for a specific command and I would almost rather the type be situated in that package and not exported.

Right now, this feels like a bit of a code smell (no judgement, I have some local branches where I have done similar things where I've created a bespoke console package and overwritten small parts to do things like this), and I'm happy we are able to make progress with the design as is.  Maybe it would feel better once we split this interface into a few smaller parts (i.e. and input half and and output half) since in that world we at least would have a type where if you call half the methods on it, it ends up exploding.

Interested if you have other thoughts about how we might want to change things up to make this less likely.  What's the key problem we are trying to solve here?

Reply:
I can make it private.  I thought it might be useful for other components in the future, but we can wait to see if that ever happen